_source/_posts/2020-12-08-beginners-guide-to-jwt.adoc

```diff
@@ -0,0 +1,175 @@
+---
+layout: blog_post
+title: "A Beginner's Guide to JWTs"
+author: brian-demers
+by: advocate
+communities: [security]
+description: "What is a JWT? How are they used? Find out in this post!"
+tags: [security, tokens, jwt]
+tweets:
+- "Learn about the format of #JWT and how they are used to create secure applications🔒"
+- "JWTs are a key part of securing the web, learn more about them in this post📡"
+- "Still confused about what a #JWT is? this post will help‼️"
+image: blog/featured/okta-java-headphones.jpg
+type: awareness
+---
+:toc: macro
+:page-liquid:
+
+JSON Web Tokens (JWT) are used everywhere (even places they shouldn't be). This post will cover the basics of what you need to know about JWT and the related specifications in the **J**avascript **O**bject **S**igning and **E**ncryption (JOSE) family.
+
+TIP: JWT is pronounced "jot".
+
+toc::[]
+
+== What is a JWT
+
+A JWT is a structured security token format used to encode JSON data. Tokens can be cryptographically verified to ensure they have not been changed. There are two types of JWTs:
+
+- **J**SON **W**eb **S**ignature (JWS)
+- **J**SON **W**eb **E**ncryption (JWE)
+
+The data in a JWS is public, meaning anyone with the token can read the data, whereas a JWE is encrypted and private.  To read data contained within a JWE, you need both the token and a secret key.
+
+Most of the time, when you JWTs are used, it's actually JWS. The 'S' (the signature) is the important part and allows the token to be validated.  For the rest of this post, I'm going to talk about JWS and only lightly touch on the differences of a JWE.
+
+== How JWTs Are Used
+
+OAuth 2.0 identity providers (IdP) commonly use JWTs for access tokens.   You may have seen an HTTP request with an authorization header that looks like this:
+
+[source,http]
+----
+Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.eyJuYW1lIjoiSm9lIENvZGVyIn0.5dlp7GmziL2QS06sZgK4mtaqv0_xX4oFUuTDh1zHK4U
```

Comment:
you can do better than "Joe Coder"! A dad joke perhaps?

Reply:
You don't like the name "Joe Coder"? 😜

Reply:
Now that I see that you're using that elsewhere in the post, I'm fine with it!